pkg/kubectl/rolling_updater.go

```diff
@@ -211,25 +215,31 @@ func (r *RollingUpdater) Update(config *RollingUpdaterConfig) error {
 		fmt.Fprintf(out, "Renaming %s to %s\n", newRc.Name, oldName)
 		return r.rename(newRc, oldName)
 	case PreserveRollingUpdateCleanupPolicy:
+		fmt.Fprintf(out, "Update succeeded. Preserving %s\n", oldName)
 		return nil
 	default:
 		return nil
 	}
 }
 
-func (r *RollingUpdater) getExistingNewRc(sourceId, name string) (rc *api.ReplicationController, existing bool, err error) {
-	if rc, err = r.c.GetReplicationController(r.ns, name); err == nil {
-		existing = true
-		annotations := rc.ObjectMeta.Annotations
-		source := annotations[sourceIdAnnotation]
-		_, ok := annotations[desiredReplicasAnnotation]
-		if source != sourceId || !ok {
-			err = fmt.Errorf("Missing/unexpected annotations for controller %s: %s", name, annotations)
+func (r *RollingUpdater) getExistingNewRc(sourceId, name string) (*api.ReplicationController, bool, error) {
+	rc, err := r.c.GetReplicationController(r.ns, name)
+	if err != nil {
+		return nil, false, err
+	}
+
+	if _, ok := rc.ObjectMeta.Annotations[DesiredReplicasAnnotation]; !ok {
+		return rc, true, fmt.Errorf("Missing %s annotation for controller %s", DesiredReplicasAnnotation, name)
+	}
+
+	// If there's no source associated with the RC, don't validate association.
```

Comment:
Hmmm. this logic needs tweaking; if the newRc *has* a sourceIdAnnotation but we didn't provide a sourceId, we should probably return an error, as that means we detected a partial rollout from an existing oldRc that the user (perhaps accidentally) didn't specify. I think the cases should be

* no sourceId provided, but newRc has sourceIdAnnotation: return error - partial rollout from oldName detected, but no starting rc specified.
* sourceId provided, but newRc doesn't have sourceIdAnnotation, or annotation doesn't match: return error - missing/unexpected annotation
* sourceId provided and matches newRc annotation: success, return rc
* sourceId not provided, and newRc has no annotation: success, return rc.